build.gradle

```diff
@@ -75,6 +75,7 @@ dependencies {
     implementation "com.sun.xml.bind:jaxb-impl:2.3.2"
     implementation "javax.activation:activation:1.1.1"
     testImplementation "junit:junit:4.12"
+    testImplementation group: 'org.springframework', name: 'spring-test', version: '5.2.2.RELEASE'
```

Comment:
I don't think it makes sense to include spring just for a simple unit test.

Reply:
If we want to test the edge cases, we need a MockHttpServletRequest that allows us to set custom headers and remote address. 
If you want I can remove the tests overall but I don't see any significant downside of adding the dependency.

Reply:
You don't need a framework to have a mock implementation of a class.

Reply:
How about creating my own mock implementation as an inner class of the test class?

Reply:
Sounds good to me.